### PR TITLE
Add example "Add actions directly", review needed

### PR DESCRIPTION
**Pull request by W4anD0eR96**
Since this tutorial shows actions in grammar only by associating a action object, I add an example to introduce write actions directly.

As I am a nonnative, maybe there is some improper expression in the text.

Thanks for your review!

### CODE REVIEW

File: doc/Language/grammar_tutorial.pod6 [hunk 1]
```diff
@@ -567,6 +567,41 @@ Here's the final code:
     }
     =end code
 
+=head2 Add actions directly
+
+Above we see how to associate grammars with a action objects and perform
```
Comment (Tyil) [inline doc/Language/grammar_tutorial.pod6]: You can leave off the "a", so it becomes "with action objects".

File: doc/Language/grammar_tutorial.pod6 [hunk 2]
```diff
@@ -567,6 +567,41 @@ Here's the final code:
...
+
+    G.parse('sub f ( a ) { }');
+    # OUTPUT: «func f␤param a␤end f␤»
+    =end code
+
+This example is a reduced portion of a parser. Let's focus more on the feature
+it shows.
+
+First, we can add actions inside the grammar ifself, and such actions are
+performed once the match arrive them. Note that action object's method will be
```
Comment (Tyil) [inline doc/Language/grammar_tutorial.pod6]: "once the match arrive them", might want to rewrite it as "arrives at them". I'm not sure if a match "arrives" at a certain action either.

File: doc/Language/grammar_tutorial.pod6 [hunk 3]
```diff
@@ -567,6 +567,41 @@ Here's the final code:
...
+performed after the whole regex item matched, while actions inside the grammar
+are more flexible. Second, it shows what C<make> really does, which is no more
+than a sugar of C<$/.made = ...>. And this trick introduces a way how we pass
+message within a regex item.
```
Comment (Tyil) [inline doc/Language/grammar_tutorial.pod6]: "how we pass messages" or "how we pass a message".

File: doc/Language/grammar_tutorial.pod6 (hunk 2 above)
Comment (W4anD0eR96) [reply doc/Language/grammar_tutorial.pod6]:
```
grammar G {
  rule TOP {
    | a { say "42" } b
    | a { say "12" } c
  }
}

G.parse("a c");
```
I want to show that both `say "42"` and `say "12"` are reached, as the example above.
How can I express this?

File: doc/Language/grammar_tutorial.pod6 [hunk 4]
```diff
@@ -567,6 +567,40 @@ Here's the final code:
...
+
+Above we see how to associate grammars with action objects and perform
+actions on the match object. However, when we want to deal with the match
+object, that isn't the only way. See the example below:
+
+    =begin code
+    grammar G {
+      rule TOP { <function-define> }
+      rule function-define {
+        'sub' <identifier> {
```
Comment (moritz) [inline doc/Language/grammar_tutorial.pod6]: I really don't like the indentation of the code block here, as it suggests a relation to the previous regex atoms that isn't there.

Most regexes/grammars I've seen or written put the curly braces of multi-line blocks on separate lines, which makes them stand out as they should.

For single-line code blocks, it's ok to put the curlies on the same line as the code block itself:

```
    { say "end " ~ $/.made; }
````

File: doc/Language/grammar_tutorial.pod6 [hunk 5]
```diff
@@ -567,6 +567,40 @@ Here's the final code:
...
+
+    G.parse('sub f ( a ) { }');
+    # OUTPUT: «func f␤param a␤end f␤»
+    =end code
+
+This example is a reduced portion of a parser. Let's focus more on the feature
+it shows.
+
+First, we can add actions inside the grammar ifself, and such actions are
+performed once the match arrives at them. Note that action object's method will
```
Comment (moritz) [inline doc/Language/grammar_tutorial.pod6]: Technically it's not the match that arrives at the actions, but rather the control flow of the regex. (The match is related to the position in the string, not directly to the position within the regex).

File: doc/Language/grammar_tutorial.pod6 (hunk 4 above)
Comment (W4anD0eR96) [reply doc/Language/grammar_tutorial.pod6]: I try to rewrite it following the style at https://github.com/moritz/perl-6-regex-code/blob/master/chapter-09-grammars/08-symbol-table.p6

BTW, looking forward to your new book!

File: doc/Language/grammar_tutorial.pod6 (hunk 5 above)
Comment (W4anD0eR96) [reply doc/Language/grammar_tutorial.pod6]: That is what I want to express

File: doc/Language/grammar_tutorial.pod6 (hunk 2 above)
Comment (geekosaur) [reply doc/Language/grammar_tutorial.pod6]: "once their [corresponding] patterns are matched"?

File: doc/Language/grammar_tutorial.pod6 [hunk 6]
```diff
@@ -567,6 +567,42 @@ Here's the final code:
...
+    }
+
+    G.parse('sub f ( a ) { }');
+    # OUTPUT: «func f␤param a␤end f␤»
+    =end code
+
+This example is a reduced portion of a parser. Let's focus more on the feature
+it shows.
+
+First, we can add actions inside the grammar ifself, and such actions are
```
Comment (Tyil) [inline doc/Language/grammar_tutorial.pod6]: "ifself" → "itself"

**Pull request merged**